--- a/qtbase/src/corelib/global/qtypetraits.h
+++ b/qtbase/src/corelib/global/qtypetraits.h
@@ -481,4 +481,11 @@ struct not_c
 // Checks whether a type is unsigned (T must be convertible to unsigned int):
+#if defined(__clang__)
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Wenum-constexpr-conversion"
+#endif
 template <typename T>
 struct is_unsigned
     : integral_constant<bool, (T(0) < T(-1))> {};
+#if defined(__clang__)
+#  pragma clang diagnostic pop
+#endif
